src/NUnitFramework/framework/TestContext.cs

```diff
@@ -369,7 +369,13 @@ public static void AddTestAttachment(string filePath, string? description = null
             if (!Path.IsPathRooted(filePath))
                 filePath = Path.Combine(TestContext.CurrentContext.WorkDirectory, filePath);
 
-            if (!File.Exists(filePath))
+            string prefix = string.Empty;
+#if NETFRAMEWORK
+            if (filePath.Length > 260)
+                prefix = @"\\?\";
```

Comment:
Whether this works or not depends on a registry setting which people (and the github runners) might not have set.
It also might need to be set in the manifest of the .exe running the tests.
See [Maximum File Path Limitation](https://learn.microsoft.com/en-au/windows/win32/fileio/maximum-file-path-limitation?tabs=registry#enable-long-paths-in-windows-10-version-1607-and-later) for details.

Reply:
It is also Windows specific, which requires a runtime test, e.g.: `OSPlatform.IsWindows10` or `OSPlatform.Platform == PlatformID.Win32NT`
For other OS there is no path length limit.

Reply:
Thanks for catching this @manfred-brands . I'd forgotten about mono runtime on Linux/Mac

Reply:
Got it. As adding prefix for long filepaths is workaround. From Windows 10 version 1607, onwards long file path can be enabled through registry and manifest file. So do we really need to add this workaround in AddTestattachment ? Though we can add this along with runtime test for platform like below-
```
            string prefix = string.Empty;
#if NETFRAMEWORK
            if (Environment.OSVersion.Platform == PlatformID.Win32NT & filePath.Length > 260)
                prefix = @"\\?\";
#endif
```

Reply:
Currently the Framework test on Linux fails because of this `prefix`:
```
System.IO.DirectoryNotFoundException : Could not find a part of the path "/home/runner/work/nunit/nunit/src/NUnitFramework/tests/bin/Release/net462/\\?\/home/runner/work/nunit/nunit/src/NUnitFramework/tests/bin/Release/net462/AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA.tmp".
```

So yes, we do need the test on OS.

If you explanation is correct that long file names only work without the prefix if explicitly enabled, I would still add this workaround if that works for everybody.

Reply:
Incorporated Comments-Added OS test to add prefix in case of win32NT.